### ISSUE TITLE
32 Bit OS X Dynlinking Broken

### ISSUE THREAD

**Issue by joelburget** [coq/coq#6658]
#### Version

master

#### Operating system

OS X 10.13.2

#### Description of the problem

Master fails to build on switch 4.06.0+32bit.

```
File "toplevel/coqtop_bin.ml", line 1:
Error: No implementations provided for the following modules:
         Dynlink referenced from kernel/kernel.cmxa(Nativelib),
           library/library.cmxa(Library), vernac/vernac.cmxa(Mltop)
```

Reproducible by both me and @herbelin. His message on gitter:

> 	`-natdynlink no` indeed fails with master using ocaml (4.06.0) 32bits (on OS X) with `No implementations provided for the following modules: Dynlink referenced from kernel/kernel.cmxa(Nativelib), library/library.cmxa(Library), vernac/vernac.cmxa(Mltop)`. Is it because of `ocamlfind` not able to find the manually built fake dynlink.cmxa?

Also (though this turned out to not work):

> I could reproduce your issue which is actually a known one. You shall find the fix in the header of file dev/dynlink.ml. The header is not informative enough: you should also install `dynlink.a` and overwrite `dynlink.cmi`. Then, I could make Coq 8.7 compilable with 4.06 32 bits on OS X, using option `-natdynlink no`.

CC @ejgallego

**Comment by herbelin** [comment]
> Also (though this turned out to not work):

See an extended fix which works for 8.8 at #7480.

**Comment by ejgallego** [comment]
It seems that support for 32bit binaries is going to be removed from OSX very soon, if I understand correctly. https://support.apple.com/en-us/HT208436

Thus we cannot do a lot on this front.
